Sources/AWSLambdaRuntimeCore/LambdaContext.swift

```diff
@@ -91,8 +99,10 @@ extension Lambda {
                       cognitoIdentity: String? = nil,
                       clientContext: String? = nil,
                       logger: Logger,
+                      tracer: TracingInstrument,
                       eventLoop: EventLoop,
-                      allocator: ByteBufferAllocator) {
+                      allocator: ByteBufferAllocator)
```

Comment:
@ktoso I think we have an abstraction that deals with combining the baggage and logger. could be appropriate here so that the logs include the tracing meta?

Reply:
Yes we have an adapter that handles the conversions etc, it's used like this:

```
import BaggageLogging

// let logger has to become:
public var logger: Logger {
  self._logger.with(context: self.baggage) // because we always want to use the "latest" value, if someone added some values there etc
}
```

Reply:
so should we use it here?

Reply:
@ktoso response from another thread

> The lambda context should conform to [LoggingBaggageContextCarrier](https://github.com/slashmo/gsoc-swift-baggage-context/blob/main/Sources/BaggageLogging/LoggingBaggageContextCarrier.swift) as well 

note that `LoggingBaggageContextCarrier` requires logger to be mutable, not just return a logger with the "latest" (baggage) context like in the snippet above

Reply:
Also I think the logger should not use the "latest" but the "current" context.

This gets tricky if the (baggage) context is propagated in a reference type.

I tried to summarize it here https://github.com/slashmo/gsoc-swift-tracing/issues/126#issuecomment-675294268